PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py

```diff
@@ -348,6 +351,10 @@ def miniAOD_customizeCommon(process):
     )
     process.deepTauIDTask = cms.Task(process.deepTau2017v2p1, process.slimmedTaus)
     task.add(process.deepTauIDTask)
+    if 'newDMPhase2v1' in tauIdEmbedder.toKeep:
```

Comment:
it would be preferable to do this with the `phase2_common` modifier (using `toModify()`, `toReplaceWith()`, etc.)

Reply:
At least for the task.add we would need a lambda in the modifier. We thought it's more compact this way and the modules should be run if this ID is 'to be kept' whether it is due to the phase2 modification or sth else. If you still prefer the modifier, please confirm and I will switch back.

Reply:
You don't, actually. You can make an object called `_deepTauIDTaskPhase2` as a clone of `deepTauIDTask`, add the additional module, and then swap them using `toReplaceWith()`. (The leading underscore in the name ensures that it won't get attached to the process otherwise.)

Reply:
Thanks, I was not aware of this possibility. I will try

Reply:
It seems like `Task` has no method `clone()`. How did you mean to clone it? On the other hand it's maybe still better to do it via a lambda. This would have one line less if I understand your plan with the cloning correctly.

Reply:
My mistake, for `Task` you have to use `copy()`, not `clone()`

Reply:
@swozniewski you can find examples a few lines below :)

Reply:
thanks. Right, I just realized that :)